lib/api_callable.js

```diff
@@ -149,107 +150,133 @@ function retryable(aFunc, retry, otherArgs) {
-          return;
-        }
-        if (settings.flattenPages) {
-          var resources = response[pageDescriptor.resourceField];
-          for (var i in resources) {
-            if (!stream.writable) {
-              break;
+      function streaming() {
+        apiCall(argument, function(err, response) {
+          if (err) {
+            stream.emit('error', err);
+            stream.end();
+            return;
+          }
+          if (!stream.writable) {
+            return;
+          }
+          if (settings.flattenPages) {
+            var resources = response[pageDescriptor.resourceField];
+            for (var i in resources) {
+              if (!stream.writable) {
+                break;
+              }
+              stream.write(resources[i]);
             }
-            stream.write(resources[i]);
+          } else {
+            stream.write(response);
+          }
+          if (!stream.writable) {
+            return;
+          }
+          var nextPageToken = response[pageDescriptor.responsePageTokenField];
+          if (!nextPageToken) {
+            stream.end();
+            return;
           }
-        } else {
-          stream.write(response);
-        }
-        if (!stream.writable) {
-          return;
-        }
-        var nextPageToken = response[pageDescriptor.responsePageTokenField];
-        if (!nextPageToken) {
-          stream.end();
-          return;
-        }
-        argument[pageDescriptor.requestPageTokenField] = nextPageToken;
-        setTimeout(streaming, 0);
-      });
+          argument[pageDescriptor.requestPageTokenField] = nextPageToken;
+          setTimeout(streaming, 0);
+        });
+      }
+      setTimeout(streaming, 0);
     }
-    setTimeout(streaming, 0);
-    return stream;
   };
 }
 
 /**
  * Converts an rpc call into an API call governed by the settings.
  *
- * In typical usage, `func` will be a callable used to make an rpc request.
- * This will mostly likely be a bound method from a request stub used to make
- * an rpc call.
+ * In typical usage, `func` will be a promsie to a callable used to make an rpc
+ * request. This will mostly likely be a bound method from a request stub used
+ * to make an rpc call. It is not a direct function but a Promise instance,
+ * because of its asynchronism (typically, obtaining the auth information).
  *
- * The result is created by applying a series of function decorators defined
- * in this module to `func`.  `settings` is used to determine which function
- * decorators to apply.
+ * The result is a function which manages the API call with the given settings
+ * and the options on the invocation.
  *
- * The result is another callable which for most values of `settings` has
- * has the same signature as the original. Only when `settings` configures
- * bundling does the signature change.
- * @param {APIFunc} func - is used to make a bare rpc call
+ * @param {Promise.<APIFunc>} func - is a promise to be used to make a bare rpc
```

Comment:
This is a promise because we are adding auth? Can we rename this to, e.g., `funcWithAuth` indicate that? Otherwise I find it hard to read lines like

```
func.then(function(func) {
   ...
}
```

because it's hard to tell the functions apart.